src/openvslam_ros.cc

```diff
@@ -36,7 +37,7 @@ void system::publish_pose() {
     nav_msgs::msg::Odometry pose_msg;
     pose_msg.header.stamp = node_->now();
     pose_msg.header.frame_id = "map";
-    pose_msg.child_frame_id = "camera_link";
+    pose_msg.child_frame_id = "odom";
```

Comment:
This is in the camera frame, see the `cam_pose_wc` this info is populated from? We need to populate a transform and use TF to back out the odom->map component of the map->camera_link transformation.

Also, the "camera_link" should use the message's header frame, not hardcoded. We should store the `camera_frame_` a a member variable in the callback functions.

Reply:
Ok. Definitely, there are some misunderstood concepts from my side. I'll continue this discussion in Slack and close it for now...

Reply:
@SteveMacenski, does this routine need to read a `camera_link->odom` transform (published by some node like robot state publisher), and then dynamically calculate&publish `map->odom` transform back. Or publishing just `map->camera_link` transform by this method will be enough for SLAM node (as I understood, TF is designed to automatically calculate the required `map->odom` transformation if all of its sub-transforms are already present in the TF-graph)?